doc/source/templates/README.md

```diff
@@ -16,79 +16,129 @@ Coming soon...
-          aws: doc/source/templates/configs/compute/cpu/aws_small.yaml
-      large:
-        compute_config:
-          # Relative to `path`
-          gcp: doc/source/templates/configs/compute/cpu/gcp_large.yaml
-          aws: doc/source/templates/configs/compute/cpu/aws_large.yaml
+    ```{note}
+    The template should be self-contained and not require any external files.
+    This requirement is to simplify the testing procedure.
     ```
 
-    Make sure that you include a small/large version for the template.
-    See the following table for a description of template size:
+2. Add another copy of the template that includes test-specific code.
 
-    | Attributes | Small-scale Version | Large-scale Version |
-    | -- | -- | -- |
-    | Number of Nodes | 1 | > 1 |
-    | Dataset size | Subset (of partitions/labels/rows) | Full example dataset |
-    | Model size | Pruned/mini version of the model | Full model |
-    | Runtime | 30-60s | Up to ~5-10 minutes |
+    NOTE: The need for a second test copy is temporary. Only one notebook will be needed
+    from 2.5 onward.
 
-    When you specify the template's compute config, see `doc/source/templates/configs` for defaults.
+    **Label all test-specific code with the `remove-cell` Jupyter notebook tag.**
+
+    **Put this test copy in `doc/source/templates/tests/test-copy.ipynb`.**
 
-3. Add a nightly release test for the template in `release/release_tests.yaml`.
+3. Add a smoke-test version of the template to run in pre-merge CI in `doc/BUILD` under the templates section.
+
+    Set the `SMOKE_TEST` environment variable, which should be used in your template to
+    set certain smoke test parameters (like limiting dataset size).
+
+    **Make sure that you tag the test with `"gpu"` if required, and any other tags
+    needed for special dependencies.
+
+    ```python
+    py_test_run_all_notebooks(
+        size = "large",
+        include = ["source/templates/tests/batch_inference.ipynb"],
+        exclude = [],
+        data = ["//doc:workspace_templates"],
+        tags = ["exclusive", "team:ml", "ray_air", "gpu"],
+        env = {"SMOKE_TEST": "1"},
+    )
+    ```
+
+4. Add a release test for the template in `release/release_tests.yaml` (for both AWS and GCE).
+
+    **Use the `release_test_cluster_env.yaml` and `*_release_test.yaml` files for cluster env / compute configs.**
+    These contain placeholders for regions and cloud ids that our CI infra will fill in.
 
     ```yaml
     - name: workspace_template_small_02_many_model_training
       group: Workspace templates
-      working_dir: workspace_templates/02_many_model_training
+      working_dir: workspace_templates/tests
       python: "3.9"
-      frequency: nightly
+      frequency: nightly-3x
       team: ml
       cluster:
         cluster_env: ../configs/release_test_cluster_env.yaml
-        cluster_compute: ../configs/compute/cpu/aws_small.yaml
+        cluster_compute: ../configs/compute/cpu/aws_release_test.yaml
```

Comment:
Out of scope of this PR, but is there an implicit requirement for all templates to have the same cluster configuration? One thing to be aware of is that maybe down the line we may need specific cluster configs per example.

Reply:
No requirement! Can use any cluster and compute config. It just happens to all be the shared ones for now.